### ISSUE TITLE
[Feature Request] Auto-update of filters

### ISSUE THREAD

**Issue by noobdesigner922** [0xCUB3/wBlock#9]
It would be really great if wBlock would 

1. show when some filter was last updated, and
2. auto-update the filters in the background.

Right now when I open wBlock, I have to click on "Update Filters" and manually select filters one by one.

**Comment by 0xCUB3** [comment]
Yep this is coming. I need to figure out background daemons so it can update filters in the background.

**Comment by arjpar** [comment]
I've been looking into this too and it does present some issues. There will have to be a different implementation for macOS & iOS/iPadOS/visionOS. macOS we can use a background service, but iOS/iPadOS/visionOS may need manual refreshing, which could include a notification sent to the user depending on the interval they set, or the user would have the keep the app open in the background in which case it may be able to auto-refresh, I'll have to double check.

**Comment by 0xCUB3** [comment]
Added in [78569d1](https://github.com/0xCUB3/wBlock/commit/78569d1a5c4fff6443d2ee91d4905bd65642458c)

**Comment by 0xCUB3** [comment]
> > No, unfortunately it requires the app to be running for now. I am working on getting it in a LaunchAgent, but I don't want to have to request too many permissions from the user.
> 
> 
> 
> Is it possible to just run it in the browser like other extensions with a built-in settings page instead of a separate app?

It would be tricky due to the limitations of safari extensions. I still think it will be possible, though.

**Comment by arjpar** [comment]
@TheKivster For content blocking you can't update it through the browser, because its converted to Content Blocking rules through the main app. For advanced blocking, it's also converted with the content blocking rules. It's not possible to do it in the browser I believe because we use SafariConverterLib to convert AdGuard format to content blocking & advanced blocking rules. We could theoretically do the advanced blocking separately, but that would incur a duplication of effort, thus it's more efficient and practical to do them all at once in the main app, rather than in the browser.

**Comment by 0xCUB3** [comment]
This feature has been added into the latest TestFlight alpha build (not release). Note that on iOS, you cannot force quit the app, as Apple won't allow background updates. On macOS, though, you can quit the main app and still get automatic filter updates! Enjoy :)

https://testflight.apple.com/join/F93erUGR
